client/app/components/application.coffee

```diff
@@ -41,7 +42,14 @@ Application = React.createClass
 
     mixins: [
         TooltipRefesherMixin
-        StoreWatchMixin [SettingsStore, RefreshesStore, RouterStore, MessageStore, LayoutStore, AccountStore]
+        StoreWatchMixin [
+            SettingsStore
+            RefreshesStore
+            RouterStore
+            MessageStore
+            LayoutStore
+            AccountStore
+        ]
```

Comment:
I've just review indent here but: I think any top-level component (aka: a component that's aware of stores changes) _should_ listen to the _main store_ (aka RouterStore) only, except if it needs to access specific data (e.g. Notifications from the NotificationStore).

Note that when we'll switch to Redux, this question won't remains as redux exposes only one store that trigger the change event :wink: